### PR TITLE
test: Avoid installing Cilium for K8sBandwidth if tests are skipped

### PR DESCRIPTION
**Pull request by pchaigno**
The overall structure for test K8sBandwidth looks to have been extracted from K8sServices. It works fine but is more complex than necessary and leads to unintended behavior when tests are skipped. This pull request simplifies the structure to have a single conditional `Context` (conditioned on net-next kernel) inside which the three `It` tests are run.

Cilium was also installed with the bandwidth manager enabled *before* the conditional `Context`. That installation would therefore happen regardless of whether bandwidth tests should actually be skipped, sometimes even leading to [flakes on 4.9 kernels](https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-K8s/3740/testReport/junit/Suite-k8s-1/16/K8sBandwidthTest_Checks_Bandwidth_Rate_Limiting/).

Removing this initial installation of Cilium implies that the test pods are now deployed (once for all tests) before Cilium is installed. We therefore need to wait for the test pods, with a new helper `waitForTestPods()`, after each re-installation of Cilium.

### CODE REVIEW

**Comment by pchaigno** [comment]
test-me-please

File: test/k8sT/Bandwidth.go
Comment (fristonio) [inline test/k8sT/Bandwidth.go]:
```suggestion
		res.ExpectSuccess("Unable to apply %s", demoYAML)
```

**Review by fristonio — APPROVED** [review]
Looks great overall. Just one minor nit, does not need a second review. :rocket: :100:

**Review by borkmann — APPROVED** [review]

**Review by kkourt — APPROVED** [review]

**Comment by pchaigno** [comment]
I've pushed the small suggestion from Deepesh. All tests were green before that, no need to re-run.

**Pull request merged**
